src/public/components/text-input.js

```diff
@@ -0,0 +1,7 @@
+import K, * as U from "karet.util"
+import React     from "karet"
+
+export const TextInput = ( { value } ) =>
+  <input type="text"
+         value={ value }
+         onChange={ U.getProps( { value } ) }/>
```

Comment:
The example in this note is not something that should be done here, but here is an example of a couple patterns I've often used with components that are basically just simple wrappers for HTML elements:

```js
export const TextInput = ( { value, mount, onChange, ...props } ) =>
   <input type="text"
          value={ value }
          onChange={ U.actions( U.getProps( { value } ), onChange ) }
          ref={ mount }
          {...props}/>
```

* This gives the possibility to customize the component by giving it additional props like `className`.
* It is also possible to give it a `mount` property that is translated to `ref` to get access to the DOM `input` element.
* The outer `onChange` is composed with the built-in action.  Sometimes similar patterns are useful for merging `className`s given as parameters.